addons/portal/wizard/portal_wizard.py

```diff
@@ -158,100 +125,88 @@ def get_error_messages(self, cr, uid, ids, context=None):
                 "- Grant access only to contacts with unique emails"))
         return error_msg
 
-    def action_apply(self, cr, uid, ids, context=None):
-        error_msg = self.get_error_messages(cr, uid, ids, context=context)
+    @api.multi
+    def action_apply(self):
+        error_msg = self.get_error_messages()
         if error_msg:
             raise UserError( "\n\n".join(error_msg))
 
-        for wizard_user in self.browse(cr, SUPERUSER_ID, ids, context):
-            portal = wizard_user.wizard_id.portal_id
-            user = self._retrieve_user(cr, SUPERUSER_ID, wizard_user, context)
-            if wizard_user.partner_id.email != wizard_user.email:
-                wizard_user.partner_id.write({'email': wizard_user.email})
-            if wizard_user.in_portal:
-                # create a user if necessary, and make sure it is in the portal group
-                if not user:
-                    user = self._create_user(cr, SUPERUSER_ID, wizard_user, context)
-                if (not user.active) or (portal not in user.groups_id):
-                    user.write({'active': True, 'groups_id': [(4, portal.id)]})
-                    # prepare for the signup process
-                    user.partner_id.signup_prepare()
-                wizard_user.refresh()
-                self._send_email(cr, uid, wizard_user, context)
-            else:
-                # remove the user (if it exists) from the portal group
-                if user and (portal in user.groups_id):
-                    # if user belongs to portal only, deactivate it
-                    if len(user.groups_id) <= 1:
-                        user.write({'groups_id': [(3, portal.id)], 'active': False})
-                    else:
-                        user.write({'groups_id': [(3, portal.id)]})
-
-    def _retrieve_user(self, cr, uid, wizard_user, context=None):
+        portal = self.wizard_id.portal_id
+        user = self.sudo()._retrieve_user()
+        if self.partner_id.email != self.email:
```

Comment:
if it's multi then you cannot use directly self.partner_id.email. need to iteration